src/Helper/ArchiveHelper.php

```diff
@@ -347,9 +357,11 @@ private function uploadDocumentToCase(string $caseId, string $webformAttachmentE
       foreach ($fileIds as $fileId) {
         /** @var \Drupal\file\Entity\File $file */
         $file = $fileStorage->load($fileId);
-
+        $filename = $file->getFilename();
+        $fileExtension = pathinfo($filename, PATHINFO_EXTENSION);
         $fileContent = file_get_contents($file->getFileUri());
-        $getOrganizedFileName = $webformLabel . '-' . $submission->serial() . '-' . $file->getFilename();
+
+        $getOrganizedFileName = substr_replace($filename, '-' . $webformLabel . '-' . $submission->serial(), strrpos($filename, '.' . $fileExtension), 0);
```

Comment:
Even the second time I don't understand this spell. Consider putting it into a function (to prevent code duplication).

Should we handle the case when it's not a PDF attachment (like in lines 338–340)?

Reply:
Added helper function converting filename into a fitting GetOrganized name.

We do indeed not need the extra case 👍  - we only allow PDF attachment.